index.html

```diff
@@ -1823,29 +1823,350 @@ <h5><code>invokeaction</code></h5>
+          id="profile-5.2.2.1-thing-protocol-binding-actions-invokeaction-9">
+            If providing an Asynchronous Action Response, a Web Thing MUST send 
+            an HTTP response with:
           </p>
+          <ul>
+            <li>Status code set to <code>201</code></li>
+            <li>
+              <code>Location</code> header set to the URL of a dynamically 
+              created <code>ActionStatus</code> resource, the
+              <a href="https://tools.ietf.org/html/rfc3986#section-3.1">URI
+              scheme</a> [[RFC3986]] of which MUST resolve to <code>http</code>
+              or <code>https</code>
+            </li>
+          </ul>
+          <pre class="example">
+            HTTP/1.1 201 CREATED
+            Location: /things/lamp/actions/fade/123e4567-e89b-12d3-a456-426655
+          </pre>
+        </section>
+
+        <section id="queryaction">
+          <h5><code>queryaction</code></h5>
+          <p>
+            A <code>queryaction</code> operation is used to query the current 
+            state of an ongoing action request.
+          </p>
+          <p class="rfc2119-assertion"
+          id="profile-5.2.2.2-thing-protocol-binding-actions-queryaction-1">
+            A Web Thing which provides <a href="#async-action-response">
+            Asynchronous Action Response</a>s to an <code>invokeaction</code>
+            operation on an <code>Action</code> MUST also support 
+            <code>queryaction</code> operations on 
+            that same <code>Action</code>.
+            A Web Thing which only provides <a href="#sync-action-response">
+            Synchronous Action Response</a>s to an <code>invokeaction</code> 
+            operation on an <code>Action</code> SHOULD NOT support 
+            <code>queryaction</code> operations on that same
+            <code>Action</code>.
+          </p>
+          <p class="rfc2119-assertion"
+          id="profile-5.2.2.2-thing-protocol-binding-actions-queryaction-2">
+            The URL of an <code>ActionStatus</code> resource to be used in a 
+            <code>queryaction</code> operation MUST be obtained from the 
+            <code>Location</code> header of an <a href="#async-action-response">
+            Asynchronous Action Response</a>.
+          </p>
+          <p class="rfc2119-assertion"
+          id="profile-5.2.2.2-thing-protocol-binding-actions-queryaction-3">
+            In order to query the status of an action request, a Consumer MUST 
+            send an HTTP request to a Web Thing with:
+          </p>
+          <ul>
+            <li>Method set to <code>GET</code></li>
+            <li>
+              URL set to the URL of the <code>ActionStatus</code> resource
+            </li>
+            <li>
+              <code>Accept</code> header set to <code>application/json
+              </code>
+            </li>
+          </ul>
+          <pre class="example">
+          GET /things/lamp/actions/fade/123e4567-e89b-12d3-a456-426655
```

Comment:
So my understanding of the current proposal is that it supports only polling to determine if an action is completed.  This is OK, and a good place to start.  We *might* want to consider a slight extension to also support long-polling, i.e. the ability to "observe" a dynamic ActionStatus resource.  Then we just have to figure out how to specify whether an immediate response or a long-polling response is requested when doing a GET (could be a query parameter on the URL, i.e. ?observe) that would indicate a response is requested only when the status changes; to avoid race conditions, perhaps the request could also provide the *current* known status, i.e. ?observe,status=running and then the system would only (generally) respond once the status changes from the provided status.   It might also need to time out, in which case it might respond even if there is no change to the status, and the consumer would have to reinvoke it in that case.  Anyway... even if we don't add this now, I can see how it would be done as a relatively simple extension of the current proposal, so I think accepting the current proposal provides a good baseline, even if it only supports regular polling for now.

Reply:
> So my understanding of the current proposal is that it supports only polling to determine if an action is completed. This is OK, and a good place to start. We _might_ want to consider a slight extension to also support long-polling

That's correct, to keep things simple. Another option we could consider is to use Server-Sent Events to be consistent with the proposed Events protocol binding in #92.

Reply:
> Then we just have to figure out how to specify whether an immediate response or a long-polling response is requested when doing a GET (could be a query parameter on the URL, i.e. ?observe)

For Server-Sent events this could be done with HTTP content negotiation, by requesting either a content type of `application/json` or `text/event-stream` in the `Accept` header.